### PR TITLE
E-879 refactor: change flow of exchange oauth token

### PR DESCRIPTION
**Pull request by dudiq**

### CODE REVIEW

**Comment by dudiq** [comment]
@lamartire ready for review

File: src/plugins/OauthPlugin/Oauth/OauthPkceStrategy.js [hunk 1]
```diff
@@ -22,45 +19,24 @@ export default class OauthPkceStrategy {
...
    * @param {object} fields
    * @return {Promise<object>}
    */
-  async exchangeCodeToToken(fields) {
-    const { payload, status, error } = await this.context.ask(
-      MESSENGER_METHODS.EXCHANGE_TOKEN_REQUEST,
-      fields,
-    );
+  async exchangeCodeToToken(oauthServer, fields) {
+    const formData = Object.keys(fields).reduce((form, key) => {
```
Comment (lamartire) [inline src/plugins/OauthPlugin/Oauth/OauthPkceStrategy.js:30]: I think `acc` is much cleaner, because in `reduce` we are always working with accumulators

Comment (dudiq) [reply src/plugins/OauthPlugin/Oauth/OauthPkceStrategy.js:30]: acc in more abstract, than form

**Comment by ifiri** [comment]
@kazurus approved

**Pull request merged**